lib/tx.py

```diff
@@ -237,3 +264,67 @@ def read_tx(self):
 
         return TxSegWit(version, marker, flag, inputs,
                         outputs, witness, locktime), double_sha256(orig_ser)
+
+
+class DeserializerAuxPow(Deserializer):
+    VERSION_AUXPOW = (1 << 8)
+
+    def read_header(self, height):
+        '''Return the AuxPow block header bytes'''
+        start = self.cursor
+        version = self._read_le_uint32()
+        if version & self.VERSION_AUXPOW:
+            # We are going to calculate the block size then read it as bytes
+            self.cursor = start
+            self.cursor += self.COIN.BASIC_HEADER_SIZE # Block normal header
+            self.read_tx() # AuxPow transaction
+            self.cursor += 32 # Parent block hash
+            merkle_size = self._read_varint()
+            self.cursor += 32 * merkle_size # Merkle branch
+            self.cursor += 4 # Index
+            merkle_size = self._read_varint()
+            self.cursor += 32 * merkle_size # Chain merkle branch
+            self.cursor += 4 # Chain index
+            self.cursor += 80 # Parent block header
+            header_end = self.cursor
+        else:
+            header_end = self.COIN.BASIC_HEADER_SIZE
+        self.cursor = start
+        return self._read_nbytes(header_end)
+
+
+class TxZcash(namedtuple("Tx", "version inputs outputs locktime")):
+    '''Class representing a transaction.'''
+
+    @cachedproperty
+    def is_coinbase(self):
+        return self.inputs[0].is_coinbase if len(self.inputs) > 0 else False
```

Comment:
@kyuupichan this transaction class is the same as `Tx` with the difference that it checks if any inputs are present (the "JoinSplit" transactions have 0 inputs). Should I leave this class or move the check to `Tx`?